.travis.yml

```diff
@@ -265,68 +265,15 @@ jobs:
-       export VERBOSE=1                                                                  && \
-       make all                                                                          && \
-       make test                                                                         && \
-       tail -n 300 ./Testing/Temporary/LastTest.log"                                        \
-      &&                                                                                    \
-      sudo docker cp -a ${TRAVIS_BUILD_DIR} mobydick:/tmp                                   \
-      &&                                                                                    \
-      sudo docker start -a mobydick
+    script: scripts/travis_fedora.sh setup openmpi latest
 
   # fedora (with gcc 10 and mpich)
   - stage: fedora
     dist: xenial
-    script: |
-      sudo docker pull registry.fedoraproject.org/fedora:rawhide                         || \
-      sudo docker pull fedora:rawhide                                                    && \
-      sudo docker create --name mobydick fedora:rawhide /bin/bash -c                        \
-      "dnf -y upgrade                                                                    && \
-       dnf -y install environment-modules git                                               \
-                      gfortran openblas-devel cmake mpich-devel make gcc-c++             && \
-       . /etc/profile.d/modules.sh                                                       && \
-       module load mpi                                                                   && \
-       cd /tmp                                                                           && \
-       cd arpack-ng                                                                      && \
-       git status                                                                        && \
-       git log -2                                                                        && \
-       mkdir -p build && cd build                                                        && \
-       cmake -DEXAMPLES=ON -DMPI=ON -DICB=ON ..                                          && \
-       export VERBOSE=1                                                                  && \
-       make all                                                                          && \
-       make test                                                                         && \
-       tail -n 300 ./Testing/Temporary/LastTest.log"                                        \
-      &&                                                                                    \
-      sudo docker cp -a ${TRAVIS_BUILD_DIR} mobydick:/tmp                                   \
-      &&                                                                                    \
-      sudo docker start -a mobydick
+    script: scripts/travis_fedora.sh setup mpich rawhide
```

Comment:
@sylvestre: endless-pain is boring-time-wasting ?! docker-run jobs are _all_ different in many slight details, no way you have one script for all !? you need that here https://github.com/opencollab/arpack-ng/blob/3dd2d5fe97ed40759b3b175d4a1c27ebcd9b0921/.travis.yml#L156 but not there https://github.com/opencollab/arpack-ng/blob/3dd2d5fe97ed40759b3b175d4a1c27ebcd9b0921/.travis.yml#L185, you need that here https://github.com/opencollab/arpack-ng/blob/3dd2d5fe97ed40759b3b175d4a1c27ebcd9b0921/.travis.yml#L159 but not there https://github.com/opencollab/arpack-ng/blob/3dd2d5fe97ed40759b3b175d4a1c27ebcd9b0921/.travis.yml#L187, you need that here https://github.com/opencollab/arpack-ng/blob/3dd2d5fe97ed40759b3b175d4a1c27ebcd9b0921/.travis.yml#L258 but not elsewhere. There are all different in many many places (apt-get install ` cmake/automake` , ` openmpi/mpich` , `  gcc\clang` if ` mpich/clang`  could be tested later, ...) : nothing to factorize here, only headache-time-wasting !?

Reply:
Well, there are ways to improve that.
The `--allow-run-as-root --oversubscribe` flags aren't needed. There are environment vars for that. That not only allows not having to patch the files, but also not having to differentiate between mpich and openmpi, as unknown env vars are ignored.
The different flags for cmake or configure can be passed via environment vars, then the overview in travis is also more useful.
It is not 100% reusable, but there could be some things that can be improved.

The [boutproject](https://github.com/boutproject/BOUT-dev/blob/master/.travis.yml) has imho a nicer interface, although they don't use docker, so things aren't that complicated. They have a [single script](https://github.com/boutproject/BOUT-dev/blob/master/.travis_script.sh), I guess for arpack-ng probably one script for each distribution would be required ...

It is more effort in the short run, but might pay off as it is easier to maintain one script rather then many copy-pasted versions :-)